pkg/controller/gateway/extensions.go

```diff
@@ -29,13 +29,98 @@ import (
 	istionetworkingv1alpha3 "github.com/cellery-io/mesh-controller/pkg/apis/istio/networking/v1alpha3"
 	"github.com/cellery-io/mesh-controller/pkg/apis/mesh/v1alpha2"
 	"github.com/cellery-io/mesh-controller/pkg/controller/gateway/resources"
+	"github.com/cellery-io/mesh-controller/pkg/meta"
 )
 
 func (r *reconciler) reconcileApiPublisherConfigMap(gateway *v1alpha2.Gateway) error {
+	configMapName := resources.ApiPublisherConfigMap(gateway)
+	configMap, err := r.configMapLister.ConfigMaps(gateway.Namespace).Get(configMapName)
+
+	if !resources.RequireConfigMap(gateway) {
+		if err == nil && metav1.IsControlledBy(configMap, gateway) {
+			err = r.kubeClient.BatchV1().Jobs(gateway.Namespace).Delete(configMapName, meta.DeleteWithPropagationBackground())
+			if err != nil {
+				r.logger.Errorf("Failed to delete config map %q: %v", configMapName, err)
+				return err
+			}
+		}
+		return nil
+	}
+
+	if errors.IsNotFound(err) {
+		desiredConfigMap, err := resources.CreateGatewayConfigMap(gateway, r.cfg)
+		configMap, err = r.kubeClient.CoreV1().ConfigMaps(gateway.Namespace).Create(desiredConfigMap)
+		if err != nil {
+			r.logger.Errorf("Failed to create ConfigMap %q: %v", configMapName, err)
+			r.recorder.Eventf(gateway, corev1.EventTypeWarning, "CreationFailed", "Failed to create ConfigMap %q: %v", configMapName, err)
```

Comment:
Change 'Failed to create ConfigMap' to 'Failed to create API publisher ConfigMap'